src/RolloutScorer/RolloutScorer/Scorecard.cs

```diff
@@ -0,0 +1,209 @@
+        {
+            string[] issueLinks = issueLinksString.Split(';');
+            if (issueLinks.Length > 0 && issueLinks[0].Length > 0)
+            {
+                return issueLinks;
+            }
+            else
+            {
+                return new string[0];
+            }
+        }
+
+        /// <summary>
+        /// Outputs the scorecard to a CSV file
+        /// </summary>
+        /// <param name="filePath">Path the CSV file should be output to</param>
+        /// <returns>Exit code (0 = success)</returns>
+        public async Task<int> Output(string filePath)
+        {
+            try
+            {
+                using (StreamWriter writer = File.CreateText(filePath))
+                {
+                    await writer.WriteLineAsync($"{Repo.Repo},{Date.Date.ToShortDateString()}");
+                    await writer.WriteLineAsync();
+                    await writer.WriteLineAsync($"Time to Rollout,Critical Issues,Issue Links,Hotfixes,Hotfix Links,Rollbacks,Rollback Links,Downtime,Failure");
+                    await writer.WriteLineAsync($"{TimeToRollout:c}," +
+                        $"{CriticalIssues},{string.Join(";", GithubIssues.Where(issue => Utilities.IssueContainsRelevantLabels(issue, Utilities.IssueLabel, Repo.GithubIssueLabel)).Select(issue => issue.HtmlUrl))}," +
+                        $"{Hotfixes},{string.Join(";", GithubIssues.Where(issue => Utilities.IssueContainsRelevantLabels(issue, Utilities.HotfixLabel, Repo.GithubIssueLabel)).Select(issue => issue.HtmlUrl))}," +
+                        $"{Rollbacks},{string.Join(";", GithubIssues.Where(issue => Utilities.IssueContainsRelevantLabels(issue, Utilities.RollbackLabel, Repo.GithubIssueLabel)).Select(issue => issue.HtmlUrl))}," +
+                        $"{Downtime:c},{Failure}");
+                    await writer.WriteLineAsync();
+                    await writer.WriteLineAsync($"Build,Link,Time to Rollout,Critical Issues,Hotfixes,Rollbacks,Downtime");
+                    foreach (ScorecardBuildBreakdown buildBreakdown in BuildBreakdowns)
+                    {
+                        await writer.WriteLineAsync($"{buildBreakdown.BuildSummary.BuildNumber},{buildBreakdown.BuildSummary.WebLink},{buildBreakdown.Score.TimeToRollout:c}," +
+                            $"{buildBreakdown.Score.CriticalIssues},{buildBreakdown.Score.Hotfixes},{buildBreakdown.Score.Rollbacks},{buildBreakdown.Score.Downtime:c}");
+                    }
+                }
+            }
+            catch (IOException e)
+            {
+                Utilities.WriteError($"File {filePath} could not be opened for writing; do you have it open in Excel?");
+                Utilities.WriteError(e.Message);
+                return 1;
+            }
+
+            return 0;
+        }
+    }
+
+    public class ScorecardBuildBreakdown
+    {
+        public BuildSummary BuildSummary { get; }
+        public Scorecard Score { get; set; }
```

Comment:
As far as I can tell, this is never set, so we don't need the setter. Immutable types are nice.
```suggestion
        public Scorecard Score { get; }
```

Reply:
Good call.